File: aat-lib/src/main/java/ch/bailu/aat_lib/service/directory/IteratorSummary.java
package ch.bailu.aat_lib.service.directory;


import ch.bailu.aat_lib.app.AppContext;
import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.gpx.InfoID;
import ch.bailu.aat_lib.util.sql.ResultSet;
import ch.bailu.foc.Foc;

public final class IteratorSummary extends IteratorAbstract {

    private GpxInformation info = GpxInformation.NULL;

    public IteratorSummary(AppContext appContext) {
        super(appContext);
        query();
    }

    @Override
    public GpxInformation getInfo() {
        return info;
    }


    @Override
    public int getInfoID() {
            return InfoID.LIST_SUMMARY;
    }

    @Override
    public void onCursorChanged(ResultSet cursor, Foc directory, String fid) {
            info = new GpxInformationDbSummary(directory, cursor);
    }

}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/directory/IteratorSummary.kt
package ch.bailu.aat_lib.service.directory

import ch.bailu.aat_lib.app.AppContext
import ch.bailu.aat_lib.gpx.information.GpxInformation
import ch.bailu.aat_lib.gpx.information.InfoID
import ch.bailu.aat_lib.util.sql.DbResultSet
import ch.bailu.foc.Foc


class IteratorSummary(appContext: AppContext) : IteratorAbstract(appContext) {
    private var info = GpxInformation.NULL

    init {
        query()
    }

    override fun getInfo(): GpxInformation {
        return info
    }

    override fun getInfoID(): Int {
        return InfoID.LIST_SUMMARY
    }

    override fun onCursorChanged(cursor: DbResultSet, directory: Foc, fileID: String) {
        info = GpxInformationDbSummary(directory, cursor)
    }
}
